_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,111 @@
+
+Feedback should be stated like this: _At some point we decided to approach things like this like that because we were convinced that's better because of reason X and Y. Don't you agree? Shall we re-evaluate?_
+
+Usually the matter is settled now. The author will probably accept the suggestion because the reviewer acted in a consulting role and took the author seriously by asking for his opinion.
+
+The author will likely feel not so pained by the rule, and decide to follow it. If not, a healthy discussion arises that might change something for the better for the whole team.
+
+As a general rule of thumb: __do not define too many rules and processes__. In a review: start with understanding the requirements, think about potential solutions also from a high-level, architectural and design-perspective. __How would you have solved the problem?__ Then look at the solution. __Find real bugs__. Only beginners focus on cosmetics (and usually miss real flaws).
+
+## Cosmetics are cosmetics
+
+Cosmetics are called cosmetics because they are __not substantial__. Bad formatting, incomplete documentation, bad naming or other trivial things should be handled as what they are: cosmetics. Don't ignore them, __good make-up can make a difference__. But as a reviewer, double-check to make sure your attitude fits the importance of the advice. Nothing is worse than bad vibes because of things that do not have a real impact.
+
+## Don't Override - Contribute
+
+It is generally a good idea for the reviewer to fix small issues during the code review. You can still point the author at the changes, but it doesn't make sense to invest the time twice. If you can fix a spelling error on the fly - why not do it?
+
+But don't change the design completely. Do not override the __substance__ of what your author wanted to do. Help the author, don't control the author.
+
+
+## Don't Only Focus On The Bad Things
+
+Try to emphasize good things about the code under review. One of the developers from my team, Peter Tackage, recently stated: _If I give compliments to someone's implementation then it can lessen the unwanted negative or criticizing aspects of a review. Something like: 'nice solution' or 'neat' when merited, can reinforce a positive mindset in reviews and communicate to the author that the reviewer is truly interested in solving the problem, rather than simply solving the problem their way_ - and I agree.
+Sometimes that's a cultural thing. Germans, like me, for instance tend to focus on problems. Maybe that is even an european issue while people from the USA are well-trained in praising other peoples sweet spots.
```

Comment:
I'm not so convinced by this paragraph. Specific cultural generalizations don't tend to make the best insights. We can be aware of cultural differences, but I think it's more meaningful to focus on the individual. After all, countries like the USA are very culturally diverse.